pkg/config/config.go

```diff
@@ -1327,3 +1322,28 @@ func (a *auth) ResolveJWKS(ctx context.Context) (string, error) {
 
 	return string(jwksEncoded), nil
 }
+
+// Retrieve the final base config to use taking into account the remotes override
+func (c *config) GetRemoteOverride(project_ref string) (overrideConfig baseConfig, overrideExist bool) {
+	// Iterate over all the config.Remotes
+	for _, remoteConfig := range c.Remotes {
+		// Check if there is one matching project_id
+		if remoteConfig.ProjectId == project_ref {
+			// Return the matching remote config and true to indicate an override exists
+			return remoteConfig, true
+		}
+	}
+
+	// If no matching remote config is found, return the base config and false
+	return c.baseConfig, false
+}
+
+func ToTomlBytes(config any) []byte {
+	var buf bytes.Buffer
+	enc := toml.NewEncoder(&buf)
+	enc.Indent = ""
+	if err := enc.Encode(config); err != nil {
+		fmt.Fprintln(os.Stderr, "failed to marshal toml config:", err)
```

Comment:
I think we want to return this error and let caller decide what to do with it.

For link, it's safe to log / ignore because it shouldn't affect the exit code. But for `DiffWithRemote`, we most likely want to bubble up the error since it indicates an irrecoverable failure.